mmv1/products/networkconnectivity/PolicyBasedRoute.yaml

```diff
@@ -0,0 +1,194 @@
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 30
+      update_minutes: 30
+      delete_minutes: 30
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+autogen_async: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'network_connectivity_policy_based_route_basic'
+    primary_resource_id: 'default'
+    vars:
+      pbr_name: 'my-pbr'
+      network_name: 'my-network'
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'network_connectivity_policy_based_route_full'
+    primary_resource_id: 'default'
+    vars:
+      pbr_name: 'my-pbr'
+      network_name: 'my-network'
+      ilb_name: 'my-ilb'
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    required: true
+    url_param_only: true
+    description: |
+      The name of the policy based route.
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    description: |
+      An optional description of this resource.
+  - !ruby/object:Api::Type::KeyValueLabels
+    name: 'labels'
+    description: |
+      User-defined labels.
+  - !ruby/object:Api::Type::String
+    name: 'network'
+    required: true
+    description: |
+      Fully-qualified URL of the network that this route applies to, for example: projects/my-project/global/networks/my-network.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'filter'
+    required: true
+    description: |
+      The filter to match L4 traffic.
+    properties:
+      - !ruby/object:Api::Type::Enum
+        name: 'protocolVersion'
+        required: true
+        description: |
+          Internet protocol versions this policy-based route applies to.
+        values:
+          - :IPV4
+      - !ruby/object:Api::Type::String  # probably could have been an enum, but it's a string in the API
+        name: 'ipProtocol'
+        default_value: "ALL"
+        description: |
+          The IP protocol that this policy-based route applies to. Valid values are 'TCP', 'UDP', and 'ALL'. Default is 'ALL'.
+      - !ruby/object:Api::Type::String
+        name: 'srcRange'
+        default_value: "0.0.0.0/0"
+        description: |
+          The source IP range of outgoing packets that this policy-based route applies to. Default is "0.0.0.0/0" if protocol version is IPv4.
+      - !ruby/object:Api::Type::String
+        name: 'destRange'
+        default_value: "0.0.0.0/0"
+        description: |
+          The destination IP range of outgoing packets that this policy-based route applies to. Default is "0.0.0.0/0" if protocol version is IPv4.
+  - !ruby/object:Api::Type::Enum
+    name: 'nextHopOtherRoutes'
+    exactly_one_of:  # next_hop union
```

Comment:
Do we need `exactly_one_of` here or `conflicts`? Like is the `next_hop` union field required?

Reply:
Yeah, this one if required- oneofs are a bit weird and can't be made required in the API schema, but can be checked at runtime. Configs w/o either fail.